tests/regression/run_regression.py

```diff
@@ -32,8 +31,8 @@
     export_missing = exports - references
     sys.exit(
         "❌ Found differences in files present: ❌\n"
-        f"Files present in reference not in export: {str(ref_missing)}\n"
-        f"Files present in export not in reference: {str(export_missing)}"
+        f"Files present in reference not in export: {ref_missing!s}\n"
+        f"Files present in export not in reference: {export_missing!s}"
```

Comment:
Is `str()` not the default here?

Reply:
str() inside fstring calls is a thing that was a constant source of complaints, i'll get the raw code back

Reply:
Yeah I'm just wondering why python even has `!s`... like without it, is there any behavior change on some edge cases or something? I'd assumed that f-string formatting would apply str() on non-strings for us, but maybe not.

Reply:
ah, ok here we go: [RUF010](https://docs.astral.sh/ruff/rules/explicit-f-string-type-conversion/)